extensions/eclipse-che-theia-terminal/src/browser/terminal-widget/remote-terminal-widget.ts

```diff
@@ -69,14 +80,27 @@ export class RemoteTerminalWidget extends TerminalWidgetImpl {
             }
         }));
 
+        const badDefaultLoginErr = 'command terminated with exit code 126';
+        const missedPrivilegesErr = 'cannot create resource "pods/exec"';
         this.toDispose.push(this.remoteTerminalWatcher.onTerminalExecError(errEvent => {
             if (this.terminalId === errEvent.id) {
                 if (this.options.closeWidgetOnExitOrError) {
                     this.dispose();
                 }
                 this.onTermDidClose.fire(this);
                 this.onTermDidClose.dispose();
+
                 this.logger.error(`Terminal error: ${errEvent.stack}`);
+                this.channel.appendLine(errEvent.stack);
+
+                let reason = '';
+                if (errEvent.stack.indexOf(badDefaultLoginErr) !== -1) {
+                    reason = ' Possible reason is bad default login.';
+                } else if (errEvent.stack.indexOf(missedPrivilegesErr) !== -1) {
+                    reason = ' Possible reason is workspace service account lacks some privileges.';
+                }
+                reason += ' See more in "Output".';
+                this.messageService.error(`Terminal failed to open.${reason}`);
```

Comment:
Put the space on the template so that you do not need to remember it on the messages.`Terminal failed to open. ${reason}`

Reply:
How about `Terminal failed to connect`?